### ISSUE TITLE
 "Sign in with GitHub to Agree" button broken on CLA pages

### ISSUE THREAD

**Issue by jimkring** [cla-assistant/cla-assistant#574]
URL to the linked Repo/Org:

https://cla-assistant.io/direct-state-transfer/
 
![image](https://user-images.githubusercontent.com/381432/83068600-74112d80-a01d-11ea-83cc-a7d21984b54a.png)

Steps to reproduce the problem:

1. Ensure not signed into GitHub
2. Visit a CLA Page
3. Click on the "Sign in with GitHub to Agree" button

What is the expected result?

I expected it to go to a sign-in page and then I could complete the CLA process.

What happens instead?

I get a 404 page.

![image](https://user-images.githubusercontent.com/381432/83068786-af136100-a01d-11ea-9519-94fb441da0c0.png)

Any other information? (attach screenshot if possible)

Note that if I first sign in to github by visiting https://cla-assistant.io and clicking the sign-in button on that page, then I'm able to complete the CLA process (since I'm already signed in)

**Comment by KharitonOff** [comment]
The reason of the issue is the url format, which CLA assistant expects. Usually we navigate users to the CLA page from open PRs. The correct form of the URL is `https://cla-assistant/<org name>/<repo name>` the repo name is a mandatory parameter. Your url works well, as long CLA assistant just shows the CLA for the linked org (repo name is not required yet). After the button click though CLA assistant tries to find the corresponding repository to update open PRs and this leads to an error, since the repo name in your case is undefined...
This means, that you just need to add any repository name from your organization to the url to fix the issue.

**Comment by pimterry** [comment]
I also just run into this exact same issue too, as did one of my contributors, and there's a few duplicate issues in this repo too.

It'd be really useful if this worked as expected!

> After the button click though CLA assistant tries to find the corresponding repository to update open PRs and this leads to an error, since the repo name in your case is undefined...

Would it be possible to make this update PRs in _every_ repo in the organization that was listed, in this case? Or if that's not possible, could we show a list of repos and ask the user to pick one? That has some issues, but it'd be much better than failing with a 404.

**Comment by KharitonOff** [comment]
> Would it be possible to make this update PRs in every repo in the organization that was listed, in this case? 

Unfortunately it is not possible due to GitHubs API rate limits... If an org has 100 repos with 100 PRs in each it would generate to high payload on both systems.

Asking user to pick the right repo could be a much better solution here. I hope we can find someone who is ready to invest his/her time to implement this improvement!

**Comment by pimterry** [comment]
In theory there is an even better option, although I'm not familiar with the internals of CLA Assistant so I don't know if it'd work in practice.

The list of outstanding unsigned PRs could be stored with the userid in mongo when they're initially opened, and then when the user logs in via github we could just update those PRs specifically, filtering to just the org in question. That way you don't need to scanning the entirety of any GitHub repos or orgs.

That's more major, but it'd solve this more completely, and use far fewer requests to the GH API. Is that possible?

**Comment by KharitonOff** [comment]
Yes, you are right and we even store this information already :) So the fix shouldn't be as difficult 👍
